src/compositions/usePrefetching.ts

```diff
@@ -1,51 +1,97 @@
-import { MaybeRefOrGetter, toValue, watch } from 'vue'
+import { MaybeRefOrGetter, onBeforeUnmount, onMounted, ref, Ref, toValue, watch } from 'vue'
 import { usePropStore } from '@/compositions/usePropStore'
 import { isWithComponent, isWithComponents } from '@/types/createRouteOptions'
-import { getPrefetchOption, PrefetchConfigs } from '@/types/prefetch'
+import { getPrefetchOption, PrefetchConfigs, PrefetchStrategy } from '@/types/prefetch'
 import { ResolvedRoute } from '@/types/resolved'
 import { isAsyncComponent } from '@/utilities/components'
+import { useVisibilityObserver } from './useVisibilityObserver'
 
 type UsePrefetchingConfig = PrefetchConfigs & {
   route: ResolvedRoute | undefined,
 }
 
 type UsePrefetching = {
+  element: Ref<HTMLElement | undefined>,
   commit: () => void,
 }
 
 export function usePrefetching(config: MaybeRefOrGetter<UsePrefetchingConfig>): UsePrefetching {
-  let props: Record<string, unknown> = {}
+  const prefetchedProps = new Map<PrefetchStrategy, Record<string, unknown>>()
+  const element = ref<HTMLElement>()
 
   const { getPrefetchProps, setPrefetchProps } = usePropStore()
+  const { observe, unobserve, isElementVisible } = useVisibilityObserver()
+
+  const commit: UsePrefetching['commit'] = () => {
+    const props = Array.from(prefetchedProps.values()).reduce((acc, value) => {
```

Comment:
Use a full word for `acc`
```suggestion
    const props = Array.from(prefetchedProps.values()).reduce((acc, value) => {
```